zebra-state/src/on_disk.rs

```diff
@@ -106,9 +140,26 @@ impl SledState {
 
         Ok(by_hash.contains_key(key)?)
     }
+
+    /// Return a utxo if it exists.
+    ///
+    /// # Details
+    ///
+    /// Returns an error if the UTXO has been spent and returns `Ok(None)` if the
+    /// UTXO is unknown.
+    ///
+    fn get_utxo(
+        &self,
+        _outpoint: OutPoint,
+    ) -> Result<Either<TransparentOutput, broadcast::Receiver<TransparentOutput>>, Error> {
+        // if contains outpoint, return outpoint, if not, check if we have a
+        // channel for it, if so subscribe and return the receiver, if not,
+        // create a channel, insert, and return the receiver
```

Comment:
I don't think we can return the UTXO until we have the previous block, because any block before the current block can spend the UTXO.